src/mixer/basetrackplayer.cpp

```diff
@@ -415,8 +416,17 @@ void BaseTrackPlayerImpl::slotLoadFailed(TrackPointer pTrack, const QString& rea
         qDebug() << "Failed to load track (NULL track object)" << reason;
     }
     m_pChannelToCloneFrom = nullptr;
+
     // Alert user.
+    // Show only one message for this track at a time in case this slot is called
+    // repeatedly for the same track while any prior message is still shown.
```

Comment:
```suggestion
    // Due to the event loop inside QMessageBox, slotLoadFailed() can be called recursively
    // via repeated loads attempts from a controller. A prior message is still shown in this case.  
    // In this case we don't show a second message box for the same track.
```

An alternative suggestion for the comment, with a bit more details. An you take it or integrate it in your comment?

Reply:
Sure, I extended the comment.
will squash if you think it's okay.